### ISSUE TITLE
The native app doesn't launch without Internet connection

### ISSUE THREAD

**Issue by oper2k** [FlutterFlow/flutterflow-issues#2950]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I created a mobile app, but it doesn't launch without an internet connection. It just gets stuck on the loading screen (white screen on iOS and logo screen on Android).
I mean the mobile apps that I created using FlutterFlow.

### Expected Behavior

The mobile app launches on the Home or Welcome screen.

### Steps to Reproduce

It’s difficult to reproduce this error; it’s possible that some widget or package doesn’t work without an internet connection. I have several such projects, and some of them launch without an internet connection.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFfl+fq2ItOsbQC7q6JbsFv+SkQHHkcTIMziNZ+exUbfaL1BqQqdc78XFVvXcOhaVNcD0yKgkQJzN7QufHXC/EnPgqdfYhl/6hiTgz0e2i6b7GVELmOWGtPBexjC1S73KWC3R1TCNNZW1o/3zqDeu6iShLDFOu/Zwh9f6PHaOI=

### Context

This started happening in one of the projects a month ago. Now, more and more projects are behaving this way and don't launch without an internet connection.

### Visual documentation

![unnamed](https://github.com/FlutterFlow/flutterflow-issues/assets/76915134/16da70ae-1a8b-41f1-879b-a7432afc5b4a)

### Additional Info

I mean the mobile apps that I created using FlutterFlow.

### Environment

```markdown
- FlutterFlow version: 4.1.53
- Platform: iOS, Android
- Browser name and version: Safari
- Operating system and version affected: macOS 14.5
```

### General

- [ ] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [ ] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [ ] ~100% of the time
- [ ] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by paulperez-dev** [comment]
Hi @oper2k! I was able to run the app on my Android device without an internet connection. An image did not load (as expected), but the app started.

Can you please verify if the issue persists? Notice that the version we use to verify this issue is the one created at the moment of the bug report code generation. Any further change is not included. So, if you changed some configurations after reporting the issue, please attach a new bug report code where the issue is present, thanks!

**Comment by oper2k** [comment]
> Hi @oper2k! I was able to run the app on my Android device without an internet connection. An image did not load (as expected), but the app started.
> 
> Can you please verify if the issue persists? Notice that the version we use to verify this issue is the one created at the moment of the bug report code generation. Any further change is not included. So, if you changed some configurations after reporting the issue, please attach a new bug report code where the issue is present, thanks!

Did you enable remote config and create a some Parameter?

**Comment by paulperez-dev** [comment]
Hi @oper2k! I run your app, with all the configuration you have there.

**Comment by paulperez-dev** [comment]
@oper2k Given the nature of the problem you're highlighting, we need a blank project with the issue steps set up in it. Without these details, it might be a bit challenging for us to investigate the issue. Please share the bug report code of the new project.

**Comment by paulperez-dev** [comment]
Hi @oper2k! Sorry for the late response. The latest shared project have some configuration issues: 

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/6a88bb46-b65e-4589-b87f-5ce492de471a/c524d1e8-baaa-45bf-a8ae-a5ebbc4bb341?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS82YTg4YmI0Ni1iNjVlLTQ1ODktYjg3Zi01Y2U0OTJkZTQ3MWEvYzUyNGQxZTgtYmFhYS00NWJmLWE4YWUtYTVlYmJjNGJiMzQxIiwiaWF0IjoxNzE3NzA0MDE5LCJleHAiOjMzMjg4MjY0MDE5fQ.bMcCIXL_d99hzVFUvX7GWlHrT4s0RKQrlxHLq-cB1rg " alt="image.png" width="293" height="248" />

So, no way to test it. Also, it would be better if you can share a bug report code from a blank project, so we can triage the issue faster, thanks!

**Comment by oper2k** [comment]
Hi
I created a new project for you but I can't to copy bug report code
I got the error 
<img width="1696" alt="Снимок экрана 2024-06-07 в 11 01 54" src="https://github.com/FlutterFlow/flutterflow-issues/assets/76915134/7d90971c-b19a-4ff9-9029-0a9a969b3536">

This is the link to my project https://app.flutterflow.io/project/test-remote-config-mnweij?tab=widgetTree&page=HomePage

You need to turn off the internet connection on mobile phone before fist run and you'll get error
<img width="1696" alt="Снимок экрана 2024-06-07 в 10 58 53" src="https://github.com/FlutterFlow/flutterflow-issues/assets/76915134/13003e3e-bd79-4f1e-953c-305149805660">

If you turned on the Internet connection the mobile app launch normally

It's working only first time launch. If you launch the application for the first time with the internet enabled, it will start normally and will continue to work without the internet. You will need to delete it and launch it again.

Steps for reproducing:
1. Download the code
2. Open project in VS code
3. Open any android simulator 
4. Turn off internet connection on simulator
5. Try to first launch app

**Comment by azizirac** [comment]
Confirm, I have the same problem. I spent all the day trying to figure out what was the issue. 

The app was working fine, the day before, and the in the morning wasn't working. I spend the entire day debugging and rolling back changes. Only after realising the phone had wifi disconnected and enabling it, then it worked. 

The logs were mentioning issues about Remote config. It was wired because IOS, web test, emulators were all working. The issue was only in my android device (Pixel 8a) which wasn't connected to internet. So I think the exception was't catch properly in the code.

To double check, I put my iPhone in plane mode, and I had same issue there too.

**Comment by paulperez-dev** [comment]
Hi @oper2k, @azizirac. This is the expected behavior of the package (it also happens inf Flutter). The first time the app lunches it needs to look for the remote configuration, but if there is no internet connection, it fails. When trying to get the remote value after the first successful launch and there is no internet, the package get the value from cache.

I'm closing this issue since this is not a bug.

**Comment by azizirac** [comment]
@paulperez-dev, I understand that Flutter has this behaviour so you are not considering as a bug. However we am using FlutterFlow to configure the remote config, and I think everyone would benefit from a grace failure instead of a blank page. I even have default value on my remote config, which I thought they are used as a fall back in case of missing connection, but looks like they are not.

If you are not considering it as a bug, can you keep open this issue as feature request? Good apps don't freeze in the launch page. User will have no idea what's going on. At least a pop-up or info banner alerting users that the app needs internet. Is there a way to achieve it?

**Comment by ready4sea** [comment]
@paulperez-dev, having encountered the same issue as the other users, I finally found this closed bug report.

While I understand that the Flutter package also "fails" in such conditions, I believe what @azizirac was saying is that the FlutterFlow use of the package could be corrected.

Indeed, at present, when launching the app while offline, the app attempts to `fetchAndActivate()`. And, as mentioned, this will fail. Fine. But the real issue is that the exception in not caught by the FlutterFlow implementation. As a result the app freezes on launch. On the other hand, with a try catch block, the app will then run as intended, as such :
![image](https://github.com/user-attachments/assets/946e6bef-3b3c-4237-b09c-ca72070eb3f4)

I'm sure the Flutter package intends the fetchAndActivate() call to be wrapped in a try catch block. It is the case in the example provided in this reply : https://github.com/firebase/flutterfire/issues/8658#issuecomment-1215061048. And it is also the case in that discussed here : https://akanksha98.medium.com/utilizing-firebase-remote-configs-within-flutter-integration-tests-4da66be4281e

In short, @leighajarett, could you please reconsider in order to make Firebase Remote Config work for your users ?

**Comment by azizirac** [comment]
@ready4sea, thanks for your input. That's exactly my point. The app should not freeze in these cases. I would love to communicate to customer that the app requires internet to function, but at the moment the app just freezes at the launch, leaving the user with a bad experience since has no idea why the app is not working,.

@paulperez-dev, these are some suggestions:
1. swallow the failure with a try catch, and let developer decide what to do when the config are missing using default values.
2. catch the exception and show popup message to customer about Internet issues.

**Comment by ready4sea** [comment]
Any news regarding this ticket ?

**Comment by leighajarett** [comment]
Thanks for following up - I will reopen and we will try and implement your proposed solution!

**Comment by ready4sea** [comment]
Yay ! 🙌 Looking forward to testing this one.

**Comment by leighajarett** [comment]
Yeah we just implemented your suggestion! Its still going through beta testing but should be in the product in the next week or two!

**Comment by ready4sea** [comment]
> Yeah we just implemented your suggestion! Its still going through beta testing but should be in the product in the next week or two!

Sounds great !

**Comment by azizirac** [comment]
@leighajarett thank you so much!! Can’t wait to test it out:)

**Comment by azizirac** [comment]
Hi @leighajarett, I have tried to launch my app Today without internet, and it opened successfully. Thank you! 

For ref, I am using https://community.flutterflow.io/c/community-custom-widgets/post/detecting-internet-connectivity-qDwV7wPZjCvIp4b to check the connection on my side and show a message to users in case. It would be nice if there is a native way in FlutterFlow, but not urgent. Thanks
